pkg/controller/controlplane/valuesprovider.go

```diff
@@ -371,16 +371,35 @@ func (vp *valuesProvider) GetControlPlaneShootChartValues(
 // GetStorageClassesChartValues returns the values for the storage classes chart applied by the generic actuator.
 func (vp *valuesProvider) GetStorageClassesChartValues(
 	_ context.Context,
-	_ *extensionsv1alpha1.ControlPlane,
+	cp *extensionsv1alpha1.ControlPlane,
 	cluster *extensionscontroller.Cluster,
 ) (map[string]interface{}, error) {
 	k8sVersionLessThan118, err := version.CompareVersions(cluster.Shoot.Spec.Kubernetes.Version, "<", "1.18")
 	if err != nil {
 		return nil, err
 	}
 
+	// Decode providerConfig
+	cpConfig := &apisaws.ControlPlaneConfig{}
```

Comment:
You could move this line in before L387. It is used only in the if statement, so there is no need to expose it in the parent scope.

```suggestion
	if cp.Spec.ProviderConfig != nil {
		cpConfig := &apisaws.ControlPlaneConfig{}
		_, _, err := vp.Decoder().Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig)
```